libbeat/processors/processor_test.go

```diff
@@ -566,3 +569,145 @@ func TestDropMissingFields(t *testing.T) {
 
 	assert.Equal(t, expectedEvent, processedEvent.Fields)
 }
+
+func TestProcessorsRun(t *testing.T) {
+	cfg := []map[string]interface{}{
+		{
+			"drop_fields": mapstr.M{
+				"fields": []string{"proc.drop"},
+			},
+		},
+		{
+			"add_fields": mapstr.M{
+				"target": "added",
+				"fields": mapstr.M{
+					"field": "value",
+				},
+			},
+		},
+	}
+	processors := GetProcessors(t, cfg)
```

Comment:
I think we need to test the full Create, Read, Update, and Delete scenarios.  I was thinking a javascript processor might be good for this.  We could test creating a new string, reading the new string, updating an existing string, and deleting an existing string, and then have the javascript processor either succeed or fail and then see if we have original event in failure case or modified event in success case.  We would want to do that for numbers, booleans, arrays and objects too.

Reply:
I think it won't be necessary once the processors list migrated to the new `EventEditor` that I'm working on, a lot is still going to change here.

EventEditor's tests will cover all operations on the `Event` including `Apply` and `Reset` of all changes made by processors. I added this test just to verify that `EventEditor` or `Event.Clone` are used and original nested maps don't change, nothing else.